pandas/core/frame.py

```diff
@@ -2348,10 +2346,15 @@ def select_dtypes(self, include=None, exclude=None):
         4   True
         5  False
         """
-        include, exclude = include or (), exclude or ()
-        if not (is_list_like(include) and is_list_like(exclude)):
-            raise TypeError('include and exclude must both be non-string'
-                            ' sequences')
+
+        # GH16855 - If either include or exclude is a non-None scalar then
+        # convert to a tuple of length 1 and continue.
+        # This allows, for example, df.select_dtypes(include='object').
+        if not is_list_like(include):
```

Comment:
Using this logic doesn't cover the case when `include` is `None` as it then gets passed through as `None`, not `[None]`, and causes a `TypeError` later on.